backend/src/handlers/teachers/handler.py

```diff
@@ -5,83 +5,32 @@
 
 from src.schemas.teachers import Teacher, TeacherWithUser, TeacherBulkCreateResponse
 from src.db import db
-
+from src.utils.security import generate_random_password, hash_password
 
 router = APIRouter()
 
 
-@router.get("/", response_model=List[TeacherWithUser])
-async def get_all_teachers(
-    first_name: Optional[str] = Query(None, alias="first_name"),
-    middle_name: Optional[str] = Query(None, alias="middle_name"),
-    last_name: Optional[str] = Query(None, alias="last_name"),
-    limit: Optional[int] = Query(10, gt=0),
-    offset: Optional[int] = Query(0, ge=0),
-    sort_by: Optional[str] = Query("lastName"),
-    order: Optional[Literal["asc", "desc"]] = Query("asc"),
-):
-
-    query: dict = {}
-    if first_name:
-        query["firstName"] = {"$regex": first_name, "$options": "i"}
-    if middle_name:
-        query["middleName"] = {"$regex": middle_name, "$options": "i"}
-    if last_name:
-        query["lastName"] = {"$regex": last_name, "$options": "i"}
-
-    sort_order = 1 if order == "asc" else -1
-    allowed_sort_fields = ["firstName", "middleName", "lastName", "createdAt"]
-    sort_field = sort_by if sort_by in allowed_sort_fields else "lastName"
-
-    cursor = (
-        db.teachers
-        .find(query)
-        .sort(sort_field, sort_order)
-        .skip(offset)
-        .limit(limit)
-    )
-
-    teachers = await cursor.to_list(length=None)
-    for t in teachers:
-        t["_id"] = str(t["_id"])
-        if "userId" in t and isinstance(t["userId"], ObjectId):
-            t["userId"] = str(t["userId"])
-
-    adapter = TypeAdapter(List[TeacherWithUser])
-    return adapter.validate_python(teachers)
-
-
 @router.post("/", response_model=TeacherWithUser, status_code=status.HTTP_201_CREATED)
 async def create_teacher(teacher: Teacher):
     try:
-        teacher_dict = teacher.model_dump(by_alias=True, exclude={"id"})
-        result = await db.teachers.insert_one(teacher_dict)
-        created = await db.teachers.find_one({"_id": result.inserted_id})
-
-        if created:
-            created["_id"] = str(created["_id"])
-            if "userId" in created and isinstance(created["userId"], ObjectId):
-                created["userId"] = str(created["userId"])
-
-        return created
-    except Exception as e:
-        raise HTTPException(status_code=500, detail=f"Database operation failed: {e}")
+        user_id, raw_pwd = await _create_linked_user(teacher)
 
+        teacher_dict = teacher.model_dump(by_alias=True, exclude={"id", "user_id"})
+        teacher_dict["userId"] = ObjectId(user_id)
 
-@router.get("/{teacher_id}", response_model=TeacherWithUser)
-async def get_teacher(teacher_id: str):
-    if not ObjectId.is_valid(teacher_id):
-        raise HTTPException(400, "Invalid teacher ID")
+        async with await db.client.start_session() as s:
+            async with s.start_transaction():
+                res = await db.teachers.insert_one(teacher_dict, session=s)
 
```

Comment:
Идея с транзакцией хорошая, но реализация требует доработке. В данном случае она не выполняет никакой роли. Инструкция insert_one итак атомарная. 

Следовало сделать так, чтобы создание user и teacher происходило в рамках одной транзакции. Например так:


```suggestion
        user_dict = await _create_linked_user(teacher) # Возвращала бы dict с user'ом
        user_id, raw_pwd = user_dict['id']

        teacher_dict = teacher.model_dump(by_alias=True, exclude={"id", "user_id"})
        teacher_dict["userId"] = ObjectId(user_id)

        async with await db.client.start_session() as s:
            async with s.start_transaction():
                res_user = await db.users.insert_one(teacher_dict, session=s)
                res_teacher = await db.teachers.insert_one(teacher_dict, session=s)

```